graylog2-server/src/main/java/org/graylog2/plugin/cluster/ClusterConfigService.java

```diff
@@ -32,6 +32,17 @@
     <T> T get(Class<T> type);
 
     /**
+     * Retrieve config payload of a certain type from the cluster configuration.
+     *
+     * Useful for custom handling of the config payload.
+     *
+     * @param type The {@link Class} of the Java configuration bean to retrieve.
+     * @param <T>  The type of the Java configuration bean.
+     * @return The payload of the requested type or {@code null} if it couldn't be retrieved.
+     */
+    <T> Object getPayload(Class<T> type);
```

Comment:
I'd rather call this method `getRawPayload()` to make it very clear, what it's doing.